v2/contribute/decisions/passwordless/0004-combined-doesSessionExist-and-validateClaims.mdx

```diff
@@ -0,0 +1,33 @@
+---
+id: "0004"
+title: Combine doesSessionExist and validateClaims into a single function
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# Combine doesSessionExist and validateClaims into a single function
+
+<DecisionHeader status="proposed" lastUpdate="2022-11-28" created="2022-11-28" deciders={["rishabhpoddar", "porcellus"]} proposedBy={["porcellus"]} />
+
+## Context and Problem Statement
+
+Apps have to call both `doesSessionExist` and `validateClaims` to completely check if the user should access a page. We want to make this simpler.
+
+## Considered Options
+
+* More documentation
+* Add function that combines both calls
+
+## Decision Outcome
+
+Add a combined function (`hasAccess`):
+- new function to web-js that combines doesSessionExists and validateClaims
+- no redirection
+- Returns: `{ hasAccess: true } | {hasAccess: false, invalidClaims: [], sessionExists: boolean}`
+
+Reasons:
+- A separate `validateClaims` is just too easy to forget.
```

Comment:
As a side note, this imples that emailverification.init also takes the mode now

Reply:
If we want to do this kind of redirection in web-js, I think this should also handle redirection if the session doesn't exist. I'll try and come up with something about that.
Either that or not handle redirection in web-js (which means we still wouldn't need the mode input)

Reply:
No no. We don't handle redirection in web-js. But the mode would tell the SDK to add the email verification claim checks automatically when the user calls `hasAccess` - so that in case email is not verified and mode is REQUIRED, then `hasAccess` will return false - as expected.

Reply:
Oh, ok, I thought I misunderstood something when we last talked about this. I agree; we shouldn't do redirections there.
Right now, the email verification validator is always added to the validators (even in the optional mode in auth-react). The difference between optional and required is the redirection only, but the check is done, and the validator appears in the `invalidClaims` array for both. I'll add a new ADR as clarification.

Reply:
Right. But we need to change that now? Cause if mode is optional, the hasAccess should return true - so we should not add the validators if mode is optional?